core/lib/types/src/debug_flat_call.rs

```diff
@@ -34,222 +32,3 @@ pub struct CallResult {
     pub output: Bytes,
     pub gas_used: U256,
 }
-
-pub fn flatten_debug_calls(calls: Vec<ResultDebugCall>) -> Vec<DebugCallFlat> {
```

Comment:
Dumb question: Would this method work here if it was supplied transaction hashes in addition to calls (i.e., `Vec<(ResultDebugCall, H256)>`), assuming that the provided calls correspond to all transactions in in a block sequentially? If so, I'd consider it cleaner than the current approach. It would also allow not loading transaction hashes in a block if it's not required to process the request (although IDK whether this would have any performance impact). Rather, they could be loaded in a separate DB request (I don't think we have it right now, but maybe I've missed it).

Reply:
The problem with the previous approach, is that it didn't server the needs to have the only method for returning traces. and it also did 2 conversions instead of 1. 
So we first converted to one type of debug calls and then to the next one